lib/signet/vm.ex

```diff
@@ -454,12 +474,69 @@ defmodule Signet.VM do
     end
   end
 
+  # Calls
+  defp static_call(context) do
+    with {:ok, context, _gas, address, args_offset, args_size, ret_offset, ret_size} <-
+           pop_call_args(context),
+         {:ok, memory_expanded, args} <-
+           Memory.read_memory(context.memory, args_offset, args_size),
+         context <- %{context | memory: memory_expanded},
+         {:ok, ffi} <- Context.fetch_ffi(context, address) do
+      case ffi.(args) do
+        {:return, return_data} ->
+          return_data_to_copy =
+            if byte_size(return_data) >= ret_size do
+              # Take left N bytes
+              <<v::binary-size(ret_size), _::binary>> = return_data
+
+              v
```

Comment:
do we really want to quietly truncate ffi return values that are bigger than expected? 🤔 

my spidey sense tingles about this, feels like maybe it should be an error. happy to be wrong tho

Reply:
This is the expected behavior in the spec for  `STATICCALL` and any other types of calls. To get a dynamically sized return value, we'll just use assembly to grab the return data and set `ret_offset` and `ret_size` to 0. We already do this in Quark.

e.g.
```
        bytes memory returnData = new bytes(returnSize);
        assembly {
            returndatacopy(add(returnData, 0x20), 0x00, returnSize)
        }

        if (!success) {
            assembly {
                revert(add(returnData, 0x20), returnSize)
            }
        }

        return returnData;
```

Reply:
Yeah, this is behavior of the EVM, generally.